File: app-java/src/main/java/net/mEmoZz/yts/java/data/bus/MovieData.java
package net.mEmoZz.yts.java.data.bus;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public class MovieData implements Parcelable {

  private long movieId;

  private String movieName;
  private String youtubeCode;

  public MovieData(long movieId, String movieName, String youtubeCode) {
    this.movieId = movieId;
    this.movieName = movieName;
    this.youtubeCode = youtubeCode;
  }

  private MovieData(Parcel in) {
    movieId = in.readLong();
    movieName = in.readString();
    youtubeCode = in.readString();
  }

  public static final Creator<MovieData> CREATOR = new Creator<MovieData>() {
    @Override
    public MovieData createFromParcel(Parcel in) {
      return new MovieData(in);
    }

    @Override
    public MovieData[] newArray(int size) {
      return new MovieData[size];
    }
  };

  public long getMovieId() {
    return movieId;
  }

  public String getMovieName() {
    return movieName;
  }

  public String getYoutubeCode() {
    return youtubeCode;
  }

  @Override public int describeContents() {
    return 0;
  }

  @Override public void writeToParcel(Parcel dest, int flags) {
    dest.writeLong(movieId);
    dest.writeString(movieName);
    dest.writeString(youtubeCode);
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/data/bus/MovieData.kt
package net.mEmoZz.yts.kotlin.data.bus

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator

/**
 * Authored by Mohamed Fathy on 08 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

class MovieData(val movieId: Long, val movieName: String, val youtubeCode: String) : Parcelable {

  constructor(parcel: Parcel) : this(
      parcel.readLong(),
      parcel.readString(),
      parcel.readString())

  override fun writeToParcel(parcel: Parcel, flags: Int) {
    parcel.writeLong(movieId)
    parcel.writeString(movieName)
    parcel.writeString(youtubeCode)
  }

  override fun describeContents(): Int {
    return 0
  }

  companion object CREATOR : Creator<MovieData> {
    override fun createFromParcel(parcel: Parcel): MovieData {
      return MovieData(parcel)
    }

    override fun newArray(size: Int): Array<MovieData?> {
      return arrayOfNulls(size)
    }
  }
}
